src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -190,22 +205,39 @@ private static void DumpProjectRestoreInfo(PackageSpec packageSpec)
                 var dgPath = Path.Combine(outputPath, $"{Guid.NewGuid()}.dg");
                 dgFile.Save(dgPath);
             }
-            catch(Exception ex)
+            catch (Exception e)
             {
-                ExceptionHelper.WriteToActivityLog(ex);
+                _logger.LogError(e.ToString());
             }
         }
 
         private static PackageSpec ToPackageSpec(ProjectNames projectNames, IVsProjectRestoreInfo projectRestoreInfo)
         {
-            var tfis = projectRestoreInfo.TargetFrameworks
+            var tfis = projectRestoreInfo
+                .TargetFrameworks
                 .Cast<IVsTargetFrameworkInfo>()
                 .Select(ToTargetFrameworkInformation)
                 .ToArray();
 
             var projectFullPath = Path.GetFullPath(projectNames.FullName);
             var projectDirectory = Path.GetDirectoryName(projectFullPath);
 
+            var originalTargetFrameworks = tfis
+                .Select(tfi => tfi.FrameworkName.GetShortFolderName())
+                .ToList();
+            var crossTargeting = originalTargetFrameworks.Count > 1;
```

Comment:
I think I see what you are saying now, I didn't realize it was being overwritten down farther. Just add some comments around this explaining that none of this is used if CPS supplied the actual data